### ISSUE TITLE
Allow "1." only ordered TOC lists output

### ISSUE THREAD

**Issue by Skylion007** [frnmst/md-toc#23]
The first feature that is blocking my implementation of the pre-commit hook is that it would be nice if we didn't have to increment the number in the markdown.

In other words instead of
```
1. FIRST ITEM
2. SECOND ITEM
3. THIRD ITEM
```
it would be nice if we could use the standard listing of 

```
1. FIRST ITEM
1. SECOND ITEM
1. THIRD ITEM
``` 
so that Github or other viewers generate the ReadMe Markdown and that adding a new section won't pollute the git blame for the other sections / line numbers.
if there is a way to generate these lists with the current args, please let me know.

**Comment by frnmst** [comment]
This is not implemented but it should be fairly easy to do

**Comment by frnmst** [comment]
I have a couple of questions:

- what would you call this option?
- should the user choose the (starting) number?

See https://spec.commonmark.org/0.28/#start-number

**Comment by Skylion007** [comment]
2. The user should be able to choose a starting number since it works on Github. 1 is by the far the most common though.
2. constant_number_lists? auto_ordered_lists? vc_friendly_ordered_lists? constant_ordered_lists? 

I am not sure what the common_mark terminology is for this feature, but it seems like the ordered lists all disregard everything after the first number and renumber themselves anyway when rendered?

**Comment by frnmst** [comment]
Ok, I'd go with `constant_ordered_lists`.

According to https://spec.commonmark.org/0.28/#start-number yes: only the first number is considered.

**Comment by frnmst** [comment]
There is a problem with nested lists:

```
1. asd
   99. o
   10. we
   117. abc
1. asd
8309. sd
```

renders this:

1. asd
   99. o
   10. we
   117. abc
1. asd
8309. sd

~

while 

```
1. asd
   1. o
   2. we
   3. abc
1. asd
8309. sd
```

renders:

1. asd
   1. o
   2. we
   3. abc
1. asd
8309. sd

I have also tried this in the commonmark interactive page: https://spec.commonmark.org/dingus/ . In that case the start integer number of the nested list must start from 1 to avoid unexpected results.

**Comment by Skylion007** [comment]
So frustrating when things don't follow the spec. I don't think starting at
1 is an awful constraint though.

**Comment by frnmst** [comment]
Yes I know. Using 1 without the ability to change it is the only way

**Comment by frnmst** [comment]
You can test the new feature in the dev branch (commit 31e80c34cde3963801f674f46b44de06382314f3).  The option is called 
`--constant-ordered-list` or  `-c` if you run:

    md_toc github -h
    md_toc redcarpet -h

**Comment by frnmst** [comment]
Feature now available in version 7.1.0

- https://github.com/frnmst/md-toc/releases/tag/7.1.0
- https://pypi.org/project/md-toc/
- https://frnmst.gitlab.io/software/#md-toc

**Comment by frnmst** [comment]
Closing. Feel free to reopen.
